frontend-ssr/redux/exercise-submissions/exercise-submissions.reducer.ts

```diff
@@ -0,0 +1,54 @@
+import { SubmissionState } from './types';
+import { EXERCISE_SUBMISSIONS } from './exercise-submissions.actions';
+
+const initialState: SubmissionState = {
+  submissions: [],
+};
+export const submissionReducer = (state = initialState, action: { type: string; payload: any;})
+: SubmissionState => {
+  switch (action.type) {
+    case EXERCISE_SUBMISSIONS.ADD: {
+      let { index, submission } = action.payload;
+      if (Array.isArray(submission)) {
+        submission = [...submission];
```

Comment:
It's already an Array, no need for this case. We only care about the one in which it's not an Array.